sklearn/cross_decomposition/_pls.py

```diff
@@ -504,6 +505,18 @@ def fit_transform(self, X, y=None):
         return self.fit(X, y).transform(X, y)
 
     # mypy error: Decorated property not supported
+    @deprecated(  # type: ignore
+        "The attribute `coef_` will be transposed in version 1.3 to be consistent "
+        "with other linear models in scikit-learn. Currently, `coef_` has a shape "
+        "of (n_features, n_targets) and in the future it will have a shape of "
+        "(n_targets, n_features)."
+    )
+    @property
+    def coef_(self):
```

Comment:
Do you want to have an estimator parameter to control the behaviour? We will need to deprecate this parameter afterwards as well. I don't know what is best.

Reply:
Can we go with "adding new parameter in 1.1 and also deprecate it immediately in 1.1"? This way we get the new behavior in 1.3 and also remove the parameter in 1.3.

Reply:
What are the options for this parameter? I would expect that you need to have `behaviour="new"` to get the new coefficient shape and silence the warning of `coef_` but it will then trigger the warning of `behaviour` so we would be back to the initial issue?

Reply:
> so we would be back to the initial issue?

I see what you mean. There would always be a warning but at different times, either during `fit` or accessing the attribute.

Is the alternative to keep the parameter for 4 versions?

- 1.1 and 1.2, `behavior="old"` saying that it will change to "new" in 1.3
- 1.3 and 1.4, `behavior="deprecated"`, "new" behavior is now default and the parameter is deprecated
- 1.5 remove

I think we've done this before and I am +0.5 for this path. (Although a lot more work for us.)

Reply:
OK let's do that then

Reply:
I'm don't think this approach solves the issue of the warnings. ``behavior="old"`` will warn that the default will change in 1.3. So in the end, we end up with warnings for 4 releases.

Is it that painful to have the warning each time we access `coef_` ? If so, we can set an attribute the first time `coef_` is accessed such that the warning is not raised for next access.

Reply:
I would be fine with this change.

Reply:
Ok I made it such that it only warn at the first access to self.coef_